src/Sylius/Behat/Context/Api/Admin/ManagingTaxCategoriesContext.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Component\Taxation\Model\TaxCategoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ManagingTaxCategoriesContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $client;
+
+    public function __construct(ApiClientInterface $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @When I delete tax category :taxCategory
+     */
+    public function iDeletedTaxCategory(TaxCategoryInterface $taxCategory): void
+    {
+        $this->client->delete('tax_category', $taxCategory->getCode());
+    }
+
+    /**
+     * @Then /^(this tax category) should no longer exist in the registry$/
+     */
+    public function thisTaxCategoryShouldNoLongerExistInTheRegistry(TaxCategoryInterface $taxCategory): void
+    {
+        Assert::false($this->isItemOnIndex('code', $taxCategory->getCode()));
```

Comment:
Add custom exception messages in true/false assertions : https://github.com/Sylius/Sylius/pull/11141#discussion_r384135754